internal/k8s/log_test.go

```diff
@@ -118,19 +135,26 @@ func TestMultipleLogWriterOptions(t *testing.T) {
 }
 
 func TestLogLevelOption(t *testing.T) {
-	log, _ := test.NewNullLogger()
+	log, logHook := test.NewNullLogger()
 	l := log.WithField("some", "field")
 	for logLevel := 1; logLevel <= 10; logLevel++ {
 		t.Run(fmt.Sprintf("log level %d", logLevel), func(t *testing.T) {
 			InitLogging(LogWriterOption(l), LogLevelOption(logLevel))
 			// Make sure log verbosity is set properly.
-			for verbositylevel := 1; verbositylevel <= 10; verbositylevel++ {
-				enabled := klog.V(klog.Level(verbositylevel)).Enabled()
-				if verbositylevel <= logLevel {
+			for verbosityLevel := 1; verbosityLevel <= 10; verbosityLevel++ {
+				enabled := klog.V(klog.Level(verbosityLevel)).Enabled()
+				if verbosityLevel <= logLevel {
 					assert.True(t, enabled)
+					klog.V(klog.Level(verbosityLevel)).Info("something")
+					klog.Flush()
+					assert.Eventually(t, func() bool { return len(logHook.AllEntries()) == 1 }, time.Second*1, time.Millisecond*10)
 				} else {
 					assert.False(t, enabled)
+					klog.V(klog.Level(verbosityLevel)).Info("something")
+					klog.Flush()
+					assert.Never(t, func() bool { return len(logHook.AllEntries()) > 0 }, time.Second*1, time.Millisecond*10)
```

Comment:
downside of this test change is that it takes the full 1s for each non-logging log level and verbosity combination to make sure no log lines come through

if that is unacceptable we could remove this check (makes the package take ~45s to run)

Reply:
Is 1s really necessary? Maybe we could start out with 500ms and increase if we need to?

Reply:
ha, yeah was able to get it down to 10ms, we'll see if it works in CI, it worked locally